### DISCUSSION TITLE
About server "support"

### DISCUSSION THREAD

**Discussion by vico93** [OreCruncher/DynamicSurroundingsFabric#40] [Q&A]
Reading the changelog of 0.0.7 on Modrinth, i got a little bit confused: if this mod is present on `mods` on a dedicated server, will it just crash the server, do nothing or do something related to the mentioned tags (like loading from a datapack)?

The line that got me confused:

> Works when connecting to Vanilla servers - yay! Includes handling impact of tag sync between server and client.

**Comment by OreCruncher** [comment]
If this mod is loaded on a Fabric server it will be ignored.  It is configured as a client only mod.  With Fabric you can connect to straight up Vanilla Minecraft servers.  Since my mod is 100% client side it could also be installed on the client.  You will get all the sound and visual effects when playing on a dedicated Vanilla server.

With this update a lot of the configuration is driven by tags from an internal datapack.  When connecting to a Vanilla server this tag configuration would be lost because Vanilla doesn't know a thing about Dynamic Surroundings.  I had to modify my processing routines to account for this by doing a side lookup into the datapacks present on the client system.

**Reply by vico93** [reply]
Oh i see. I was confused about the server-side support due to your mention of datapack stuff that, at least from what i know, is 100% on server-side.

Perharps moving those files to resourcepack since its client-side would be more aligned to how vanilla is structured.

**Reply by OreCruncher** [reply]
Understandable.  The core change I made is to use the tags feature of datapacks to mark up entities, blocks, and items.  My tags give queues to logic when processing effects.  Normally these tags would be flushed from the registry  when connecting to a server since the server is not aware of these tags.  I had to work around this issue.  Luckily Fabric has some logic that handles this for me - I just needed to use it.

Now that I mention it - a server could define dsurround tags in its datapacks, and thus can control how the mod works.  I have to think on that a bit - could be useful.
